source/fab/util.py

```diff
@@ -127,18 +147,67 @@ class CompiledFile(object):
     A Fortran or C file which has been compiled.
 
     """
-    def __init__(self, analysed_file, output_fpath: Path):
+    def __init__(self, input_fpath, output_fpath,
+                 source_hash=None, flags_hash=None, module_deps_hashes: Dict[str, int] = None):
         """
-        :param analysed_file:
+        :param input_fpath:
             The file that was compiled.
         :param output_fpath:
-            The object file.
+            The object file that was created.
+
+        """
+        # todo: Should just be the input_fpath, not the whole analysed file
+        self.input_fpath = Path(input_fpath)
+        self.output_fpath = Path(output_fpath)
+
+        self.source_hash = source_hash or 0
+        self.flags_hash = flags_hash or 0
```

Comment:
Check that hashing function can't produce `0` from data input.

Reply:
It's only 0 when we're not using a hashing function at all, as in the case of the C compiler (or tests), so there can be no clash.

Reply:
We will need to support incremental building of C code as well. I'm just mentioning this in case that had slipped through the cracks. It doesn't have to be done in this change though.

Reply:
Agreed. Of course when we do they won't default to 0.